### ISSUE TITLE
DataTable: event.preventDefault() not working in cell edit events

### ISSUE THREAD

**Issue by rpspall** [primefaces/primereact#4339]
### Describe the bug

Looking to selectively prevent DataTable cell editing based on cell / row contents.

event.preventDefault() appears to be broken in these Column element callback events:

**onCellEditInit** and **onBeforeCellEditShow**, e.g.

```
onCellEditInit = (e) => {
        let { rowData, newValue, field, originalEvent: event } = e;

// some data analysis that may require preventing editing
....
// Want to now prevent cell editing
        event.preventDefault();
}
```

Similarly broken for onBeforeCellEditShow

### PrimeReact version

9.3.1

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

**Comment by melloware** [comment]
You can now `return false;` or `originalEvent.preventDefault()` to stop the editor from displaying.

```js
const onBeforeCellEditShow = (e) => {
        let { rowData, newValue, field, originalEvent: event } = e;

        // do not show the cell editor just for the price field
        if (field === 'price') {
            event.preventDefault();
        }
};
```

or

```js
const onBeforeCellEditShow = (e) => {
        let { rowData, newValue, field, originalEvent: event } = e;

        // do not show the cell editor just for the price field
        if (field === 'price') {
            return false;
        }
};
```

**Comment by rpspall** [comment]
Thank you... When will new Release be available for "npm update" containing this and your fix to the previous bug?

**Comment by RastogiNishant** [comment]
@melloware Is there any specific planned date for the release(which can contain this bugfix)

**Comment by melloware** [comment]
> 

See my statement above.  I don't work for PrimeTek just an open source contributor so I have not insight into release schedules.

**Comment by melloware** [comment]
@RastogiNishant actually this fix was in 9.4.0 and 9.5.0 is already out so this fix is already available.
